src/HashingLogic.ts

```diff
@@ -40,6 +40,73 @@ export interface IAttestationData {
   version: string
 }
 
+export interface IAttestationType {
+  /**
+   * The type of attestation (phone, email, etc.)
+   */
+  type: keyof typeof AttestationTypeID
+  /**
+   * Optionally identifies service used to perform attestation
+   */
+  provider?: string
+  /**
+   * Attestation type nonce
+   */
+  nonce: string
+}
+
+export interface IRevocationLinks {
+  /**
+   * Hex string to identify this attestation node in the event of partial revocation
+   */
+  local: string
+  /**
+   * Hex string to identify this attestation in the event of revocation
+   */
+  global: string
+}
+
+export interface IAuxSig {
+  /**
+   * Hex string containing subject's auxiliary signature
+   * Signs the ordered stringified object containing
+   * { dataHash: hashAttestation(IAttestationData), typeHash: hashAttestation(IAttestationType)}
+   */
+  signedHash: string
+  /**
+   * Nonce to conceal unwanted revealing of aux public key
+   */
+  nonce: string
+}
+
+export interface IAttestation {
+  data: IAttestationData
+  type: IAttestationType
+  /**
+   * aux either contains a hash of IAuxSig or just a padding node hash
+   */
+  aux: string
```

Comment:
Maybe this should be optional?  Padding hash could give confusing result re: is there a signature that someone is unwilling to provide vs. is there no signature

Reply:
I think we want it to be ambiguous

Reply:
although I supposed it's possible to hide an alt data node as this aux node. to fix this I added dataHash and typeHash to the link node as a local checksum